addrmgr/addrmanager.go

```diff
@@ -738,27 +738,46 @@ func (a *AddrManager) reset() {
 	}
 }
 
+// ParseHost deconstructs a given host address to its []byte representation
+// and also returns the network address type. If the host cannot be decoded,
+// then an unknown address type is returned without error.
+func ParseHost(host string) (NetAddressType, []byte) {
```

Comment:
Alright, for those following along, we discussed this a bit offline and the overall approach this is aiming for is to make it so that there is a unique encoding per address type such that the encoding has the information necessary to reconstruct the address it represents.  It is essentially the mechanism the `addrv2` message uses.

There are pros and cons to the approach, but since a fair amount of work has already been done towards this approach we'll continue down this path.

One thing I would highly suggest though is changing the comment to make it more clear what it's actually doing.  It reads as if it's converting it to an IP `[]byte` slice which isn't really the intent.

Perhaps "... attempts to identify the given host address as a supported network address, converts it to a unique encoding based that type, and returns the result.  If the ....."

Reply:
Perhaps even calling it something more like `EncodeHost` or `HostToNetAddr` would help to clarify it even further.

Reply:
Updated, thanks!